src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -1665,16 +1650,87 @@ export class StudyViewPageStore {
             });
             return filterAttributes;
         },
-        default: [],
-        onResult: (result) => {
-            let charts = _.reduce(result, (acc, next) => {
-                acc[getClinicalAttributeUniqueKey(next)] = true;
-                return acc;
-            }, {} as {[uniqueId:string]:boolean});
-            this.changeChartsVisibility(charts);
-        }
+        default: []
+    });
+
+    readonly defaultVisibleAttributesClinicalDataCountData = remoteData<ClinicalDataCountItem[]>({
+        await: () => [this.defaultVisibleAttributes],
+        invoke: async () => {
+            return internalClient.fetchClinicalDataCountsUsingPOST({
+                clinicalDataCountFilter: {
+                    attributes: _.filter(this.defaultVisibleAttributes.result, attr => attr.datatype === 'STRING').map(attr => {
+                        return {
+                            attributeId: attr.clinicalAttributeId,
+                            clinicalDataType: attr.patientAttribute ? 'PATIENT' : 'SAMPLE'
+                        } as ClinicalDataFilter
+                    }),
+                    studyViewFilter: this.defaultFilters
+                } as ClinicalDataCountFilter
+            });
+        },
+        onResult: (data) => {
+            this.initialClinicalDataCountCharts(data);
+        },
+        default: []
+    });
+
+    readonly defaultVisibleAttributesClinicalDataBinCountData = remoteData<DataBin[]>({
+        await: () => [this.defaultVisibleAttributes],
+        invoke: async () => {
+            return internalClient.fetchClinicalDataBinCountsUsingPOST({
+                dataBinMethod: 'STATIC',
+                clinicalDataBinCountFilter: {
+                    attributes: _.filter(this.defaultVisibleAttributes.result, attr => attr.datatype === 'NUMBER').map(attr => {
+                        return {
+                            attributeId: attr.clinicalAttributeId,
+                            clinicalDataType: attr.patientAttribute ? 'PATIENT' : 'SAMPLE',
+                            disableLogScale: false
+                        } as ClinicalDataBinFilter
+                    }),
+                    studyViewFilter: this.defaultFilters
+                } as ClinicalDataBinCountFilter
+            });
+        },
+        onResult: (data) => {
+            this.initialClinicalDataBinCountCharts(data);
+        },
+        default: []
     });
 
+    @action
+    initialClinicalDataCountCharts(data: ClinicalDataCountItem[]) {
```

Comment:
i think all action methods should have verbs in name.  like, initializeClinicalDataCountCharts